xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -15165,3 +15206,123 @@ glusterd_replace_old_auth_allow_list(char *volname)
 out:
     return ret;
 }
+
+/* Adding a new node to the hashtable consisting of the volname as the key and
+ * the volinfo as the data */
+
+int32_t
+glusterd_add_volume_to_hashtable(xlator_t *this, char *volname,
+                                 glusterd_volinfo_t *volinfo)
+{
+    int32_t ret = -1;
+    uint32_t hash = 0;
+    struct cds_lfht *tmp_ht = NULL;
+    glusterd_conf_t *priv = NULL;
+    glusterd_volinfo_node *tmp_node = NULL;
+
+    GF_ASSERT(this);
+
+    priv = this->private;
+    GF_ASSERT(priv);
+
+    tmp_node = GF_MALLOC(sizeof(glusterd_volinfo_node), gf_gld_mt_hash_node);
+    if (!tmp_node) {
+        gf_smsg(THIS->name, GF_LOG_ERROR, errno, GD_MSG_NO_MEMORY, NULL);
+        goto out;
+    }
+
+    tmp_ht = priv->ht;
+    if (!tmp_ht) {
+        /*Creating the hash table, if not already created */
+        tmp_ht = cds_lfht_new(1, 1, 0, CDS_LFHT_AUTO_RESIZE, NULL);
+        if (!tmp_ht)
+            goto out;
+        priv->ht = tmp_ht;
+    }
+
+    /* Computing hash using the volname as the key */
+    hash = (uint32_t)XXH64(volname, strlen(volname), 0);
+    if (!hash) {
```

Comment:
hash cannot be 0?

Reply:
I don't think it can be.
Well, I had this doubt and did some search all gave negative answers only.

Reply:
AFAIK, a hash function can't fail. Also, based on the implementation of the hash, I don't see why it cannot be 0. We cannot do that.

Reply:
Okay